### DISCUSSION TITLE
autoConnect does not pick up the environment variables when running on Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by arvinoids** [leafsphp/leaf#151] [Q&A]
Hello, 

This is probably a configuration issue on my server or something...
My code runs fine on the development machine (Windows, using `php leaf serve`). However, when I cloned the code to my Ubuntu apache server, I get the error `"SQLSTATE[HY000] [1698] Access denied for user 'root'@'localhost'"` when autoConnect is executed. In the **Environment & details** section, I can see that the database information and credentials are all picked up from the `.env` file correctly, but for some reason, autoConnect reverts to the `'root'@'localhost'` credentials.  Does anyone have an idea what is going on? p.s. the error only comes up when I invoke a route that calls a controller, regardless if the controller access a model or not (thus, accessing / does not throw this error).

@mychidarko Great work on leaf API!

Thanks, 
Arvin

**Comment by mychidarko** [comment]
Thanks @arvinoids. Can you try grabbing any env variable using Leaf's `_env()` variable. Let's start debugging from there to see if this works

**Comment by arvinoids** [comment]
Hello @mychidarko !
Thanks for looking into this. 

This is my code: 
```php
app()->get("/getvars", function () {
    response()->json(["APP_NAME"=>_env("APP_NAME")]);
});
```

and this is what I got: 
```json
{"APP_NAME":"LEAF_API"}
```
...which to say, it does work.

**Comment by mychidarko** [comment]
That's quite weird then. I'll check the db module. Which version of Leaf Db do you have installed?

**Reply by arvinoids** [reply]
I'm on `Leaf API v3.0`. 

Also, just in case you need to know, here's my apache conf:
```apache
DocumentRoot /var/www/html

<Directory /var/www/html>
    Options Indexes FollowSymLinks
    AllowOverride All
</Directory>

SSLEngine on
SSLCertificateFile /etc/ssl/certs/apache-selfsigned.crt
SSLCertificateKeyFile /etc/ssl/private/apache-selfsigned.key
```

Thanks!

**Reply by arvinoids** [reply]
Sorry, I misunderstood. How do I know which Db version I am using? It came with the Leaf 3 API installation.

**Reply by mychidarko** [reply]
Check the composer.json file and see if there's a `leafs/db` key

**Reply by arvinoids** [reply]
Ok, here it is: `"leafs/db": "^2.0",`

**Reply by mychidarko** [reply]
Can you try reinstalling Leaf db? After that try running your app again and let's see if that fixes your issue

**Reply by arvinoids** [reply]
Sorry for my delayed response. I tried running the project on a different distro just to test (did a composer install, so vendor folder is fresh). I found out that the result is the same. 
I tried your suggestion of  uninstalling leafs/db and reinstalling it. I got the same response. 

Is there any additional setting that needs to be done on a linux/apache setup? I have not tried it with nginx as I am not very familiar with that product.

**Reply by mychidarko** [reply]
No, that's fine. I'll try to simulate your Linux environment and see what's going wrong.

**Reply by mychidarko** [reply]
In the mean time, you can use `connect` instead of `autoConnect`.

**Reply by arvinoids** [reply]
>No, that's fine. I'll try to simulate your Linux environment and see what's going wrong.

Thank you!

> In the mean time, you can use `connect` instead of `autoConnect`.

Pardon my ignorance, but I am not exactly sure how to do this since I am not calling `autoConnect` explicitly. 
As an example, I tried this code:
In my ```routes/index.php```
```php
app()->get('/test','TestsController@index');
```
and in my ```controllers/TestsController.php```
```php
 public function index()
    {
        response()->json([
            'message' => 'Hello World'
        ]);
    }
```
... and then got the error ```"SQLSTATE[HY000] [1698] Access denied for user 'root'@'localhost'"```
which I think is weird because the code does not even interact with the database.

**Reply by mychidarko** [reply]
That shouldn't happen since you're not doing anything database related

**Reply by mychidarko** [reply]
Is the code you're using on GitHub, I can take a look and see what's happening

**Reply by arvinoids** [reply]
Hello, sorry I was out for the weekend. 

No, I do not have my code on github. That is fine. I am alternating between Leaf and Lumen, and in the meantime Lumen is doing what I need. I will try to get it to work on Leaf when I have more time. Regards, /arvin

**Reply by arvinoids** [reply]
Hi @mychidarko ! I would like to ask one simple question and maybe I could have some ideas where to dig deeper into my issue. 
Let me tell you what I did for testing step-by-step:
1) I ran ```composer create-project leafs/api testproject```
2) I ran ```php leaf g:con Test```
3) I went to ```app/controllers/TestsController.php``` and found that the default index method simply returns a json, so I took it as is.
4) I went to ```app/routes/index.php``` and added a route:
          ```app()->get( 'test' , 'TestsController@index' );```
5) I ran ```php -S 0.0.0.0:5000``` on the server.
6) I opened / from the browser and got the ```{"message":"Congrats!! You're on Leaf API"}``` message.
7) I opened /test and got the ```"SQLSTATE[HY000] [1045] Access denied for user 'LEAF_DB_USERNAME'@'localhost' (using password: NO)"```

I have not configured .env, nor the database, since in this simple test I did not intend to use it. So I guess my question is, why is it trying to connect to a database? 

This is not an urgent concern, just trying to learn how Leaf works. Happy weekend!

**Reply by mychidarko** [reply]
I'll have to look into this @arvinoids
